src/modules/users/CMakeLists.txt

```diff
@@ -78,7 +88,9 @@ calamares_add_plugin(users
     SHARED_LIB
 )
 
-calamares_add_test(userspasswordtest SOURCES TestPasswordJob.cpp SetPasswordJob.cpp LIBRARIES ${CRYPT_LIBRARIES})
+if(NOT HAS_CRYPT_GENSALT)
+    calamares_add_test(userspasswordtest SOURCES TestPasswordJob.cpp SetPasswordJob.cpp LIBRARIES ${CRYPT_LIBRARIES})
```

Comment:
What makes the test unsuitable if this function is available?

If it's about testing internal functions, see comments below about how I would prefer keeping `make_salt()` around.

Reply:
I suppose the question becomes how to test it, or rather, how to make it reproducible. When a null pointer is passed to the `prefix` argument of `crypt_gensalt()`, it will pick the best available hashing method, which could potentially change from system to system (this is probably unlikely), or simply over time as newer, better methods are discovered.

Reply:
I assume the CI system has libxcrypt if it's at all using a recent distribution. Considering that `crypt_gensalt` is supposed to return random output I'm not sure what the best way to wire this up for tests is. I can only think of creating a static yescrypt or sha512crypt prefix and using that to determine if the password hashing function returns expected output?

Reply:
Aha, so we can check "does it return the thing" but since it is supposed to return the **best** thing at any given time, that thing may change and we'd end up with broken-by-external-changes tests. Not what we want. 

Thanks for the explanation.